File: Search/hashing/sum_pair.cpp

#include <bits/stdc++.h> 
  
using namespace std; 
  
void printPairs(int arr[], int arr_size, int sum) 
{ 
    unordered_set<int> s; 
    for (int i = 0; i < arr_size; i++) 
    { 
        int temp = sum - arr[i]; 
  
        if (temp >= 0 && s.find(temp) != s.end()) 
            cout << "Pair with given sum " << sum << 
                 " is (" << arr[i] << ", " << temp << 
                 ")" << endl; 
  
        s.insert(arr[i]); 
    } 
} 
  
/* Driver program to test above function */
int main() 
{ 
    int A[] = {1, 4, 45, 6, 10, 8}; 
    int n = 16; 
    int arr_size = sizeof(A)/sizeof(A[0]); 
  
    // Function calling 
    printPairs(A, arr_size, n); 
  
    return 0; 
} 

File: Search/hashing/2_Sum/sum_pair.py
def print_pairs(arr, sum):
  s = set()

  for i in range(0, len(arr)):
    temp = sum - arr[i]
    if temp >= 0 and temp in s:
      print("Pair with the given sum is", arr[i], "and", temp)
    s.add(arr[i])


if __name__ == "__main__":
  arr = [1, 4, 45, 6, 10, 8]
  n = 16
  print_pairs(arr, n)
